tests/tracking/_tracking_service/test_utils.py

```diff
@@ -371,6 +373,22 @@ def test_set_tracking_uri_with_path(tmp_path, monkeypatch, absolute):
         assert get_tracking_uri() == path.absolute().resolve().as_uri()
 
 
+def test_set_tracking_uri_update_trace_provider():
```

Comment:
Can we add a 'real' test case where mlflow auto-configure the trace_provider for different tracking_uris, and make sure get_trace/search_traces don't break